# Pérez Angeles Adrián 

Evalúa el polinomio $$x^{4} +x^{3} +2x^{2} -x$$ en x = 1.1. Utiliza variables para
evitar teclear varias veces el valor de x

In [27]:
print(" Programa para evaluar el polinomio en x=1.1")
x=1.1
i=x**4
j=x**3
k=2*x**2
print("el valor del polinomio es: ", i+j+k-x)

 Programa para evaluar el polinomio en x=1.1
el valor del polinomio es:  4.115100000000002


Da con el mayor detalle posible el resultado de evaluar las siguientes
expresiones:
(2+3)*1+2

(2+3)*(1+2)

1/2/4.0

4**0.5

4.0**(1.0/2)+1/2.0

3e3/10

10/5e-3 +1

True ==True != False

1<2<3<4<5

palindromo ="abcba"

(4*"<"+ palindromo +">"*4)*2

"abalorio" <"abecedario"

124 < 13

"124"< "13"

str(2.1)+ str(1.2)

str(2+3)

str(int (2.1)+float(3))

In [31]:
(2+3)*1+2 #se evalua como (2+3)*1+2=(5)+2=7

7

In [28]:
(2.0+3.0)*(1.0+2.0) #se evalua (2+3)*(1+2)=5*3=15 y son de tipo flotante

15.0

In [29]:
1/2/4.0 #se evalua 1/2/4=0.5/4=0.125 y es de tipo flotante debido al 4.0

0.125

In [30]:
4**0.5 #4^{1/2}=sqrt(4)=2

2.0

In [32]:
4.0**(1.0/2)+1/2.0

2.5

In [33]:
3e3/10

300.0

In [34]:
10/5e-3+1

2001.0

In [35]:
True ==True != False



True

In [36]:
1<2<3<4<5


True

In [44]:
palindromo ="abcba"
(4*"<"+ palindromo +">"*4)*2


'<<<<abcba>>>><<<<abcba>>>>'

In [39]:
124 < 13


False

In [40]:
"124"< "13"


True

In [41]:
str(2.1)+ str(1.2)


'2.11.2'

In [42]:
str(2+3)


'5'

In [43]:
str(int (2.1)+float(3))

'5.0'

3. Evalúa las siguientes expresiones, primero en papel y luego en el intérprete
de Python.

a) int(exp(2*log(3))) $$e^{2log(3)}=e^{log(3^{2})}=e^{log(9)}=2.596702959842573074$$
como marcamos la expresion la parte entera, entonces $int(e^{log(9)})=2$

b) round(4*sin(3*pi/2)) $$4sin(\frac{3\pi}{2})=4(-1)=-4$$
y como roud redondea el valor a el mas cercano y lo regresa en flotante entonces $round(4sin(\frac{3\pi}{2}))=-4$

c) abs(log10(0.01)*sqrt(25)) $$|log_{10}(0.01) \sqrt(25)|=|log_{10}(0.01)\pm 5|=|log_{10}(0.01)|5=|-2|5=10$$ ya que abs() es el valor absoluto 

In [57]:
#a)
from math import exp, log10 #esto es para diferenciar ya que python toma en automatico log como ln
int(exp(2*log10(3)))

2

In [64]:
#b)
from math import *
round(4*sin(3*pi/2))

-4

In [66]:
#c)
abs(log10(0.01)*sqrt(25))

10.0

4. Escribe un programa que almacene en una lista los números del 1 al 10
y los muestre, en orden inverso, separados por comas.

In [85]:
l=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
#l[-1], l[-2], l[-3], l[-4], l[-5], l[-6], l[-7], l[-8], l[-9], l[-10]
l[::-1]

[10, 9, 8, 7, 6, 5, 4, 3, 2, 1]

5. Escribe un programa que almacene el abecedario en una lista, elimine de
ésta las letras que ocupen posiciones que sean múltiplos de 3 y muestre
la lista resultante

In [73]:
l=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l" ,"m", "n", "ñ", "o", "p", "q", "r", "s", "t", "u", "v", "x", "y", "z"]
print("la lista es: ", l[0:]) 
l.remove("d")
l.remove("g")
l.remove("j")
l.remove("m")
l.remove("o")
l.remove("r")
l.remove("u")
l.remove("y")
print("la lista borrando los multiplos de 3 es: ", l[0:]) 

la lista es:  ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'ñ', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'x', 'y', 'z']
la lista borrando los multiplos de 3 es:  ['a', 'b', 'c', 'e', 'f', 'h', 'i', 'k', 'l', 'n', 'ñ', 'p', 'q', 's', 't', 'v', 'x', 'z']


6. Escribe un programa que pida al usuario una palabra y muestre si es un
palíndromo.

In [83]:
x=str(input('introduce una palabra:'))
print("es palindorma si True, no es palindroma si False.", x==x[::-1])

introduce una palabra:hola
es palindorma si True, no es palindroma si False. False


7. Pide números y mételos en una lista, cuando el usuario meta un 0 ya
dejaremos de insertar. Por último, muestra los números ordenados de
menor a mayor.

In [13]:
l=[]
while True:
    x= int(input("Escribe un numero: "))
    if (x==0):
        print("adios")
        break
    else:    
        l.append(x)
l.sort()
l[::]

Escribe un numero: 234235
Escribe un numero: 23
Escribe un numero: 534
Escribe un numero: 2
Escribe un numero: 4534
Escribe un numero: 76
Escribe un numero: 0
adios


[2, 23, 76, 534, 4534, 234235]

8. Escribe un programa que pida al usuario una palabra y muestre un
conteo de cada vocal.

In [1]:
while True:
    cont=0
    x=str(input("escribe una oración o escriba 0 para salir: "))
    if (x== '0'):
        print("adios")
        break
    else:
      for caracter in x:
          if caracter in 'aeiouAEIOUáéíóúÁÉíÓÚ':
              cont+= 1
              print(caracter)
      print('Hay: ',cont,'vocales en: ',x)

escribe una oración o escriba 0 para salir: Hola mundo
o
a
u
o
Hay:  4 vocales en:  Hola mundo
escribe una oración o escriba 0 para salir: 0
adios


9. Pide una frase por teclado, mete las palabras en una lista sin espacios.

In [41]:
cadena = input("escribe una frase: ") #split
print(cadena, sep='')
lista = cadena.split()
print(lista)

escribe una frase: Hola mundo
Hola mundo
['Hola', 'mundo']


10. Pide una cadena por teclado, mete los caracteres en una lista sin espacios.

In [45]:
cadena = input("Dame una cadena: ")
 
print(cadena)
 
lista = []
 
for c in cadena:
    if(c != ' '):
        lista.append(c)
 
print(lista, sep='', end='')

Dame una cadena: Hola mundo
Hola mundo
['H', 'o', 'l', 'a', 'm', 'u', 'n', 'd', 'o']

11. Haz un programa que pida el nombre de una persona y lo muestre 1000
veces, pero dejando un espacio de separación entre cada copia del nombre.

In [42]:
nombre=input("introduce un nombre: ")
 
for i in range(1000):
    print(i+1, nombre, end=" ")

introduce un nombre: Adrián
1 Adrián 2 Adrián 3 Adrián 4 Adrián 5 Adrián 6 Adrián 7 Adrián 8 Adrián 9 Adrián 10 Adrián 11 Adrián 12 Adrián 13 Adrián 14 Adrián 15 Adrián 16 Adrián 17 Adrián 18 Adrián 19 Adrián 20 Adrián 21 Adrián 22 Adrián 23 Adrián 24 Adrián 25 Adrián 26 Adrián 27 Adrián 28 Adrián 29 Adrián 30 Adrián 31 Adrián 32 Adrián 33 Adrián 34 Adrián 35 Adrián 36 Adrián 37 Adrián 38 Adrián 39 Adrián 40 Adrián 41 Adrián 42 Adrián 43 Adrián 44 Adrián 45 Adrián 46 Adrián 47 Adrián 48 Adrián 49 Adrián 50 Adrián 51 Adrián 52 Adrián 53 Adrián 54 Adrián 55 Adrián 56 Adrián 57 Adrián 58 Adrián 59 Adrián 60 Adrián 61 Adrián 62 Adrián 63 Adrián 64 Adrián 65 Adrián 66 Adrián 67 Adrián 68 Adrián 69 Adrián 70 Adrián 71 Adrián 72 Adrián 73 Adrián 74 Adrián 75 Adrián 76 Adrián 77 Adrián 78 Adrián 79 Adrián 80 Adrián 81 Adrián 82 Adrián 83 Adrián 84 Adrián 85 Adrián 86 Adrián 87 Adrián 88 Adrián 89 Adrián 90 Adrián 91 Adrián 92 Adrián 93 Adrián 94 Adrián 95 Adrián 96 Adrián 97 Adrián 98 Adrián 9

12. Escribe un programa que guarde en una variable el diccionario 'Euro':
25.82, 'Dolar': 24.113, 'Yen': 0.2189,'Dolar Canadiense': 16.7842 y
le pregunte al usuario por una divisa y muestre su valor de cambio o
un mensaje de aviso si la divisa no está en el diccionario. Completa con
todas las divisas que puedas.

In [12]:
moneda={}
moneda=dict()
moneda["Euro"]=13.82
moneda["Dolar"]=24.113
moneda["Yen"]=0.2189
moneda["Dolar Canadiense"]=16.7842
moneda["Libra"]=27.55
moneda["Real"]=3.92
moneda["Dolar Australiano"]=14.86
moneda["Franco"]=21.73

x=input("¿Que divisa deseas consultar? Dolar, Dolar Canadiense, Libra, Real, Dolar Australiano, Franco, Yen, Euro: ")


if x=='Euro':
    print(moneda.get('Euro'))
elif x=='Dolar':
    print(moneda.get('Dolar'))
elif x=='Dolar Canadiense':  
    print(moneda.get('Dolar Canadiense'))
elif x=='Libra':
    print(moneda.get('Libra'))
elif x=='Real':
    print(moneda.get('Real'))
elif x=='Dolar Australiano':
    print(moneda.get('Dolar Australiano'))
elif x=='Franco':
    print(moneda.get('Franco'))
elif x=='Yen':
    print(moneda.get('Yen'))
else: 
    print("opcion no registrada en el menú, da una opción valida. ")


¿Que divisa deseas consultar? Dolar, Dolar Canadiense, Libra, Real, Dolar Australiano, Franco, Yen, Euro: Real
3.92


13. Escribe un programa que pregunte al usuario su nombre, edad, dirección
y teléfono y lo guarde en un diccionario.

In [20]:
datos={}
datos=dict()
nom=input("¿Cuál es tu nombre?")
datos={'Nombre':nom}
edad=input("¿Cuál es tu edad? en años ")
datos['Edad']=edad
direccion=input("¿Cuál es tu dirección? ")
datos['Dirección']=direccion
print(datos)

¿Cuál es tu nombre?Adrián Pérez
¿Cuál es tu edad? 25
Cuál es tu dirección? Higuera 28
{'Nombre': 'Adrián Pérez', 'Edad': '25', 'Dirección': 'Higuera 28'}


14. Escribe un programa que almacene el diccionario con las califícaciones
de las asignaturas de un semestre 'Algebra': 10, 'Física': 9, 'Calculo':
8,'Computacion': 10, 'Geometria': 9, y, después, muestre por pantalla
las califícaciones de cada asignatura en el formato asignatura tiene
calificacion , donde asignatura es cada una de las asignaturas del
curso, y calificacion son sus califícaciones. Al fínal, debe mostrar también el número promedio del curso.

In [30]:
asignatura={'Algebra': 10, 'Física': 9, 'Calculo': 8,'Computacion': 10, 'Geometria': 9}
calificacion=0
materias=0
for x, y in asignatura.items():
    print(x, "tiene calificaión: ",y)
    calificacion+= y
    l=[x]
    len(l)
    materias+= len(l)
prom=calificacion / materias
print("El promedio de las calificaciones es: ", prom)

Algebra tiene calificaión:  10
Física tiene calificaión:  9
Calculo tiene calificaión:  8
Computacion tiene calificaión:  10
Geometria tiene calificaión:  9
El promedio de las calificaciones es:  9.2


15. Diseña un programa que pida el valor de los tres lados de un triángulo
y calcule el valor de su área y perímetro. Recuerda que el área A de un
triangulo puede calcularse a partir de sus tres lados a,b y c: así $A =\sqrt{s(s- a)(s - b)(s - c)}$ donde $s = \frac{(a + b + c)}{2}$. Esta expresión es la
fórmula de Herón de Alejandría.

In [38]:
from math import sqrt
a=float(input("ingrese el valor de el lado a: "))
b=float(input("ingrese el valor de el lado b: "))
c=float(input("ingrese el valor de el lado c: "))

perimetro= a+b+c
s= perimetro / 2
area= sqrt(s *(s-a) *(s-b) *(s-c))
print("su área es: ", area, "y su perimetro es: ", perimetro)

ingrese el valor de el lado a: 3
ingrese el valor de el lado b: 2
ingrese el valor de el lado c: 3
su área es:  2.8284271247461903 y su perimetro es:  8.0


16. Un satélite se encuentra en una órbita circular alrededor de la Tierra,
tal que completa un círculo en T segundos; T es el periodo de la órbita:

i) Muestra que la altitud h sobre la superfície de la Tierra será:
$$h=(\frac{GMT^{2}}{4 \pi^{2}})^{\frac{1}{3}}- R$$
donde G es la constante de gravitación universal y tiene un valor
$G = 6.67x10^{11} m^{3}kg^{-1}s^{-2}$, M es la masa de la Tierra con valor
$M = 5.97x10^{24}kg$ y R es el radio de la Tierra con valor $R = 6371 km$.

Recordando la tercera ley de kepler 
$$T^{2}=\frac{4 \pi^{2} }{G M}a^{3}$$ 
Donde a es el semi-eje mayor. Por lo que $a=R+h$ entonces 
$$T^{2}= \frac{4 \pi^{2}}{GM}(R+h)^{3}$$ 
por lo que despejando $R+h$ es:
$$\frac{T^{2}GM}{4 \pi^{2}}^{\frac{1}{3}}=R+h$$
Por lo que 
$$h=(\frac{GMT^{2}}{4 \pi^{2}})^{\frac{1}{3}}- R$$


(ii) Escribe un programa que le pregunte al usuario el periodo que desea
que tenga su satélite girando alrededor de la Tierra y que le regrese la
altitud, en metros, que debería tener.


(iii) ¿Qué altitud debería tener el satélite si queremos que orbite la Tierra
con un periodo de un día (órbita geosíncrona), 90 minutos, 45 minutos?
Qué observas?

(iv) En práctica, una órbita geosíncrona es aquella cuya órbita tiene un
periodo de un día sideral, que es de 23.93 horas, no de 24 horas. Por
qué es esto? ¿Cuál sería la diferencia en altitud entre T= 23.93 horas y
T= 24 horas?

In [21]:
#ii)
from math import *
G=6.67e11
M=5.97e24
R=6371000

T=float(input("Este programa calcula la altitud de un satelite a partir del periodo en segundos \n ¿Cual es el periodo?"))
h=((G*M*T)/4*pi**2)**(1. /3. ) - R
print("su altitud es h=", h, "m")


Este programa calcula la altitud de un satelite a partir del periodo en segundos 
 ¿Cual es el periodo?91800
su altitud es h= 96618621768290.73 m


iii)
para un dia y 90 min $h= 96618621768290.73 m$
para un dia y 45 min $h= 95661938850853.2 m$

iv) porque en dia sideral es lo que toma en dar una vuelta completa la tierra, mientras el dia solar se considera la rotacion completa de la tierra mas la diferencia de traslacion de la tierra para que quede en la misma poción del sol en la tierra pasado un día. El dia sideral tiene una altura de $h= 94 593 583 217 540.62 m$ mientras el dia solar tiene una altura de $h= 94685728436827.69 m$ y hay una diferencia de alturas de $\triangle h=92,145,219,287.07m$